src/views/clusteroverview/OverviewTab/resources-inventory-card/hooks/useResourcesQuantities.ts

```diff
@@ -0,0 +1,74 @@
+import { useMemo } from 'react';
+
+import {
+  modelToGroupVersionKind,
+  NodeModel,
+  TemplateModel,
+  VirtualMachineModelGroupVersionKind,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { NetworkAttachmentDefinitionModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console/models/NetworkAttachmentDefinitionModel';
+import { TEMPLATE_TYPE_LABEL } from '@kubevirt-utils/resources/template';
+import { K8sResourceCommon, useK8sWatchResources } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseResourcesQuantities = (namespace: string) => {
+  loaded: boolean;
+  networks: number;
+  nodes: number;
+  vms: number;
+  vmTemplates: number;
+};
+
+const useResourcesQuantities: UseResourcesQuantities = (namespace) => {
+  const watchedResources = {
+    vms: {
+      groupVersionKind: VirtualMachineModelGroupVersionKind,
+      namespaced: !!namespace,
+      namespace,
+      isList: true,
+    },
+    vmTemplates: {
+      groupVersionKind: modelToGroupVersionKind(TemplateModel),
+      namespaced: !!namespace,
+      namespace,
+      isList: true,
+      selector: {
+        matchExpressions: [
+          {
+            key: TEMPLATE_TYPE_LABEL,
+            operator: 'Exists',
+          },
+        ],
+      },
+    },
+    nodes: {
+      groupVersionKind: modelToGroupVersionKind(NodeModel),
+      namespaced: !!namespace,
+      namespace,
+      isList: true,
+    },
+    nads: {
+      groupVersionKind: NetworkAttachmentDefinitionModelGroupVersionKind,
+      namespaced: !!namespace,
+      namespace,
+      isList: true,
+    },
+  };
+
+  const resources = useK8sWatchResources<{ [key: string]: K8sResourceCommon[] }>(watchedResources);
+
+  return useMemo(() => {
+    const resourceQuantities = Object.keys(resources).reduce(
+      (acc, key) => {
+        acc[key] = resources[key]?.data?.length;
+        return acc;
+      },
+      { loaded: false, networks: 0, nodes: 0, vms: 0, vmTemplates: 0 },
+    );
+
+    resourceQuantities.loaded = Object.values(resources).every((resource) => resource?.loaded);
+
+    return resourceQuantities;
+  }, [resources]);
+};
```

Comment:
```suggestion
  return useMemo(() => {
    return Object.entries([key, values]).reduce(
      (acc, key) => {
        acc[key] = values?.data?.length;
        acc["loaded"] = acc["loaded"] ? acc["loaded"] && values?.loaded : values?.loaded
        return acc;
      },
      { loaded: false, networks: 0, nodes: 0, vms: 0, vmTemplates: 0 },
    );
  }, [resources]);
};
```

something like that?

Reply:
Much cleaner, thanks. I short-circuited the logic to:
`acc.loaded = acc.loaded && value.loaded`
since the only combination that would make `acc.loaded` true is if both values are true.

Reply:
can it be possible resources is only 1 field object? if so, I'm guessing only acc.loaded && value.loaded isn't enough, but if guaranteed that resources have at least 2 fields, it will be ok..